### ISSUE TITLE
Running HaaS in Dev-mode with postgresql as database backend

### ISSUE THREAD

**Issue by SahilTikale** [CCI-MOC/haas#557]
We need to update HACKING.rst with instructions to run HaaS in dev mode with postgresql database backend.
Current instructions are incomplete and needs elaboration. 
After following the given instruction I had to troubleshoot it further.

**Comment by kylehogan** [comment]
I have this documentation. Will double check then make a pull request so people can edit it.

**Comment by SahilTikale** [comment]
Did you end up with the error `command 'gcc' failed with exit status 1` while trying to do
`pip install psycopg2` in dev-mode

**Comment by SahilTikale** [comment]
you can create the PR, I will take a look at it.

**Comment by naved001** [comment]
@Izhmash you run HaaS in devmode with postgresql? Can you see if there's something in the documentation that you can add that would be helpful for others?

**Comment by Izhmash** [comment]
I'm using HIL with postgres locally, yeah.  I'll try to remember how I set it up and add any comments.

**Comment by zenhack** [comment]
The process is pretty much the same as for production -- install postgress and psycopg2, create a postgres db, and put the url in hil.cfg. you can put it in testsuite.cfg to run the tests against postgres.

The error @SahilTikale mentioned above looks like the c toolchain isn't installed, which will cause problems with any Python library that calls C.

I see the value in having extra docs around this, but I also don't want to clutter docs/ with stuff where there is nothing hil-specific going on.

Maybe just the high level I wrote above, plus a link to a wiki article?

**Comment by Izhmash** [comment]
The only thing extra I noticed when configuring for Postgresql testing was that I needed a database called 'hil_tests'.  I remember one of the tests specifically was trying to access a db with that name, and when I tried to point testsuite.cfg at my database named 'hil' it didn't work.  That part got me hung up a bit.

**Comment by zenhack** [comment]
Huh, I haven't seen that -- my local postgres db is called hil. Which test was it?

**Comment by Izhmash** [comment]
I can't remember which test, but it might have something to do with how Travis installs the databases (https://github.com/CCI-MOC/hil/search?utf8=%E2%9C%93&q=hil_tests&type=).

But anyway I was trying to figure out if it was at all possible to run the tests with a different DB name, and it sounds like you can.

**Comment by zenhack** [comment]
Yeah, it should work with any name -- hil_tests is just what we happen to use on travis.

**Comment by naved001** [comment]
> The process is pretty much the same as for production -- install postgress and psycopg2, create a postgres db, and put the url in hil.cfg. you can put it in testsuite.cfg to run the tests against postgres

@SahilTikale Does this still warrant a documentation update? 

I am gonna close this issue because the consensus here seems like we don't need an update in the docs.

 if you still think this is pertinent, please re-open it.
